### PR TITLE
Simplify cluster installation

### PR DESCRIPTION
**Pull request by boubaker**
In this PR, several modifications has been made to simplify installation :
* Replace old CLUSTER variables by new ones to simplify cluster installation: currently in one single variable, you can list all cluster nodes
* Two variables are added to allow specifying the storage type to use for binary files for JCR and Files API
* Fix definition of Connection Pools that didn't replaces the `minIdle` and `maxIdle` while the variables `EXO_DB_POOL_*_INIT_SIZE` are documented to do so
* Delete useless cluster configuration that overrides default behavior of eXo Platform
* Delete [deprecated and useless](https://tomcat.apache.org/tomcat-8.5-doc/config/listeners.html#JMX_Remote_Lifecycle_Listener_-_org.apache.catalina.mbeans.JmxRemoteLifecycleListener) configuration of JMX Remote access listener
* Allow to define chat in standalone mode, and especially to define a flow between eXo and Chat Servers that doesn't pass by Nginx (Frontend)
* Replace/Define a `jvmRoute` using node name when cluster is enabled
* Add a configuration in `context.xml` configuration file to disable session cookie appending in URL
* Avoid waiting for mongodb availability when chat is installed in standalone mode

### CODE REVIEW

File: bin/setenv-docker-customize.sh [hunk 1]
```diff
@@ -235,6 +236,8 @@ else
 
   # Update JCR datasource settings
   xmlstarlet ed -L -u "/Server/GlobalNamingResources/Resource[@name='exo-jcr_portal']/@initialSize" -v "${EXO_DB_POOL_JCR_INIT_SIZE}" \
+    -u "/Server/GlobalNamingResources/Resource[@name='exo-jcr_portal']/@minIdle" -v "${EXO_DB_POOL_IDM_INIT_SIZE}" \
+    -u "/Server/GlobalNamingResources/Resource[@name='exo-jcr_portal']/@maxIdle" -v "${EXO_DB_POOL_IDM_INIT_SIZE}" \
```
Comment (bourasom) [inline bin/setenv-docker-customize.sh:240]: Typo: it should be JCR not IDM

File: bin/setenv-docker-customize.sh [hunk 2]
```diff
@@ -243,6 +246,8 @@ else
 
   # Update JPA datasource settings
   xmlstarlet ed -L -u "/Server/GlobalNamingResources/Resource[@name='exo-jpa_portal']/@initialSize" -v "${EXO_DB_POOL_JPA_INIT_SIZE}" \
+    -u "/Server/GlobalNamingResources/Resource[@name='exo-jpa_portal']/@minIdle" -v "${EXO_DB_POOL_IDM_INIT_SIZE}" \
```
Comment (bourasom) [inline bin/setenv-docker-customize.sh:249]: Typo: it should be JPA not IDM

File: bin/setenv-docker-customize.sh [hunk 3]
```diff
@@ -376,53 +397,8 @@ else
...
-    # JCR configuration
-    add_in_exo_configuration "gatein.jcr.config.type=cluster"
-    # TODO allow to customize this
-    add_in_exo_configuration "gatein.jcr.index.changefilterclass=org.exoplatform.services.jcr.impl.core.query.ispn.LocalIndexChangesFilter"
-
-  fi
-
   # JMX configuration
   if [ "${EXO_JMX_ENABLED}" = "true" ]; then
-    # insert the listener before the "Global JNDI resources" line
```
Comment (bourasom) [inline bin/setenv-docker-customize.sh:415]: this properties are in use for us. Please keep them

File: bin/setenv-docker-customize.sh [hunk 4]
```diff
@@ -694,14 +672,6 @@ if [ -f /etc/exo/chat.properties ] && [ "${EXO_CHAT_SERVER_STANDALONE}" = "false
   sed -i 's/^chatPassPhrase=.*$/chatPassPhrase='"$(tr -dc '[:alnum:]' < /dev/urandom  | dd bs=4 count=6 2>/dev/null)"'/' /etc/exo/chat.properties
 fi
 
-
-# -----------------------------------------------------------------------------
-# Configure the eXo profiles for clustering if needed
-# -----------------------------------------------------------------------------
-if [ "${EXO_CLUSTER}" = "true" ]; then
```
Comment (bourasom) [inline bin/setenv-docker-customize.sh:701]: what would activate the profile cluster in the future?

File: bin/setenv-docker-customize.sh (hunk 1 above)
Comment (boubaker) [reply bin/setenv-docker-customize.sh]: Fixed

File: bin/setenv-docker-customize.sh (hunk 2 above)
Comment (boubaker) [reply bin/setenv-docker-customize.sh]: Fixed

File: bin/setenv-docker-customize.sh (hunk 3 above)
Comment (boubaker) [reply bin/setenv-docker-customize.sh:415]: I know that it's used by you and all others (devs, ops, community ...). In fact this configuration is deprecated in Tomcat and will be removed as [mentioned here](https://tomcat.apache.org/tomcat-8.5-doc/config/listeners.html#JMX_Remote_Lifecycle_Listener_-_org.apache.catalina.mbeans.JmxRemoteLifecycleListener)

In addition, in PRD 6.1 we will upgrade to Tomcat 9, so I will revert this until we 'll do the upgrade in product and let you re-work it.

(Note: The new option to use is `-Dcom.sun.management.jmxremote.rmi.port=XYZ`, see https://tomcat.apache.org/tomcat-9.0-doc/monitoring.html)

Comment (boubaker) [reply bin/setenv-docker-customize.sh:415]: Done

File: bin/setenv-docker-customize.sh (hunk 4 above)
Comment (boubaker) [reply bin/setenv-docker-customize.sh:701]: This option is interpreted in product : 
https://github.com/Meeds-io/meeds/blob/d9bcded2b38371342ddba3b634bc777e61065e85/packaging/plf-tomcat-resources/src/main/resources/bin/setenv.sh#L97

**Comment by boubaker** [comment]
@bourasom is that ok for you ?

**Review by bourasom — APPROVED** [review]
LGTM

**Pull request merged**